### DISCUSSION TITLE
In Nuxt has something like "Suspense" from Nextjs?

### DISCUSSION THREAD

**Discussion by astrO1** [nuxt/nuxt#30121] [Questions]
In Nextjs has a feature called Suspense when the component is loaded on the back and after that it shows on the front with the full HTML.

In Nuxt we have **https://nuxt.com/docs/api/components/client-only** **<ClientOnly>**

That works the same but the component is loaded in the client. I'm asking this because i want a good SEO for the component and want to user load the first page fast and not wait the full page.

Here the content from Nextjs

https://nextjs.org/docs/app/building-your-application/routing/loading-ui-and-streaming

**Comment by oneminch** [comment]
Vue has an experimental [`<Suspense>`](https://vuejs.org/guide/built-ins/suspense.html) component which Nuxt uses under the hood to implement some things. But, I'm not aware of a Next.js-like implementation of `<Suspense>` in Nuxt.

The way you achieve the same type of functionality in Nuxt is different from the Next.js approach. You would typically rely on composables like `useAsyncData`.

**Reply by astrO1** [reply]
I understand but i want the suspense because i dont want to show the request in my network tab

And in my experience the server component has some strange errors

**Reply by oneminch** [reply]
I may be wrong but `useAsyncData` (with `$fetch`) is supposed to help you achieve that if you are using SSR. It may not be obvious in dev mode, but using those composables prevents data re-fetching on the client. It also provides a `server` option (which is `true` by default) to ensure data is fetched on the server.

**Reply by astrO1** [reply]
The problem is that my requests in Nuxt it showing on my network tab, but in Nextjs in a server component is not showing. The Suspense from Next is like they make the request on server side and after that it show on front page the full html.

i will send a example
![Captura de tela 2024-12-04 193300](https://github.com/user-attachments/assets/c4d49582-7cc4-41e7-9389-808a249f8a61)

**Reply by oneminch** [reply]
Perhaps you should explore [server components](https://roe.dev/blog/nuxt-server-components/) and see if it solves your issue.

- [Exploring Server Components in Nuxt](https://masteringnuxt.com/blog/exploring-server-components-in-nuxt)
